kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/HttpClientUtils.java

```diff
@@ -285,41 +105,103 @@ private static boolean isIpAddress(String ipAddress) {
         return ipMatcher.matches();
     }
 
-  /**
-   * OkHttp wrongfully detects >JDK8u251 as {@link okhttp3.internal.platform.Jdk9Platform} which enables Http2
-   * unsupported for JDK8.
-   *
-   * @return true if JDK8 is detected, false otherwise-
-   * @see <a href="https://github.com/fabric8io/kubernetes-client/issues/2212">#2212</a>
-   */
-  private static boolean shouldDisableHttp2() {
-      return System.getProperty("java.version", "").startsWith("1.8");
-  }
-
-  static List<Interceptor> createApplicableInterceptors(Config config) {
-    List<Interceptor> interceptors = new ArrayList<>();
+  public static Map<String, io.fabric8.kubernetes.client.http.Interceptor> createApplicableInterceptors(Config config, HttpClient.Factory factory) {
+    Map<String, io.fabric8.kubernetes.client.http.Interceptor> interceptors = new LinkedHashMap<>();
+    
     // Header Interceptor
-    interceptors.add(chain -> {
-      Request request = chain.request();
-      if (Utils.isNotNullOrEmpty(config.getUsername()) && Utils.isNotNullOrEmpty(config.getPassword())) {
-        Request authReq = chain.request().newBuilder().addHeader("Authorization", Credentials.basic(config.getUsername(), config.getPassword())).build();
-        return chain.proceed(authReq);
-      } else if (Utils.isNotNullOrEmpty(config.getOauthToken())) {
-        Request authReq = chain.request().newBuilder().addHeader("Authorization", "Bearer " + config.getOauthToken()).build();
-        return chain.proceed(authReq);
+    interceptors.put(HEADER_INTERCEPTOR, new Interceptor() {
+      
+      @Override
+      public void before(BasicBuilder builder, HttpHeaders headers) {
+        if (Utils.isNotNullOrEmpty(config.getUsername()) && Utils.isNotNullOrEmpty(config.getPassword())) {
+          builder.header("Authorization", basicCredentials(config.getUsername(), config.getPassword()));
+        } else if (Utils.isNotNullOrEmpty(config.getOauthToken())) {
+          builder.header("Authorization", "Bearer " + config.getOauthToken());
+        }
+        if (config.getCustomHeaders() != null && !config.getCustomHeaders().isEmpty()) {
+          for (Map.Entry<String, String> entry : config.getCustomHeaders().entrySet()) {
+            builder.header(entry.getKey(),entry.getValue());
+          }
+        }
+        if (config.getUserAgent() != null && !config.getUserAgent().isEmpty()) {
+          builder.setHeader("User-Agent", config.getUserAgent());
+        }
       }
-      return chain.proceed(request);
     });
     // Impersonator Interceptor
-    interceptors.add(new ImpersonatorInterceptor(config));
+    interceptors.put("IMPERSONATOR", new ImpersonatorInterceptor(config));
```

Comment:
Maybe move the name to the interceptor class, similarly to what's done for `TokenRefreshInterceptor`?